api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_online.py

```diff
@@ -56,6 +56,22 @@ def finalize(
         return batches_u, batches_var, all_u, all_var
 
 
+class MeanAccumulator:
+    def __init__(self, n_samples: int, n_variables: int):
+        if n_samples <= 0:
+            raise ValueError("No samples provided - can't calculate mean.")
+
+        self.u = np.zeros(n_variables, dtype=np.float64)
+        self.n_samples = n_samples
+
+    def update(self, var_vec: npt.NDArray[np.int64], val_vec: npt.NDArray[np.float32]) -> None:
+        for col, val in zip(var_vec, val_vec):
```

Comment:
@bkmartinjr To the best of my knowledge, this part of the mean calculation could still benefit from numba. Is there any specific numpy function you had in mind that could avoid the loop?

Reply:
Yep, as written it will be slow.  You can either:
* put that code in a numba.jitted function and call it.  Numba will make it fast
* or, rewrite to use Numpy advanced indexing
the latter is:

> self.u[var_vec] = self.u[var_vec] + val_vec